### PR TITLE
APIGOV-18453 - unhealthy status when job or health check fail

### PR DESCRIPTION
**Pull request by jcollins-axway**

### CODE REVIEW

**Review by sbolosan — APPROVED** [review]
just curious... i mean i like the stop/start Log func... but it didn't really add anything right?

**Comment by jcollins-axway** [comment]
> just curious... i mean i like the stop/start Log func... but it didn't really add anything right?

It did. The detached interval job is a different jobType. By using those common funcs I did not have to duplicate all the logic in the start/stop funcs of an interval job to the detached interval job.

**Comment by sbolosan** [comment]
> > just curious... i mean i like the stop/start Log func... but it didn't really add anything right?
> 
> It did. The detached interval job is a different jobType. By using those common funcs I did not have to duplicate all the logic in the start/stop funcs of an interval job to the detached interval job.

Ah...  i see it now... I like it.

File: pkg/agent/periodicstatusupdate.go [hunk 1]
```diff
@@ -20,6 +22,12 @@ func (su *periodicStatusUpdate) Ready() bool {
 	if runStatusUpdateCheck() != nil {
 		return false
 	}
+	// Do not start until status will be running// get the status from the health check and jobs
```
Comment (dfeldick) [inline pkg/agent/periodicstatusupdate.go:25]: what happened here? 2 lines of comments get merged into 1 line accidentally?

Comment (jcollins-axway) [reply pkg/agent/periodicstatusupdate.go:25]: looks like it, will fix

**Review by dfeldick — APPROVED** [review]

**Review by vivekschauhan — APPROVED** [review]

**Review by tsjohns9 — APPROVED** [review]

**Pull request merged**